tools/src/ABIChecker/ABIChecker.cs

```diff
@@ -0,0 +1,156 @@
+
+            Console.WriteLine($"File Count : {baseDllFiles.Length} == {targetDllFiles.Length}");
+
+            ABITestResult directoryResult = ABITestResult.None;
+            directoryResult |= CheckChangedModule(baseDllFiles, targetDllFiles);
+            Console.WriteLine($"Module Check : {directoryResult.ToString()}");
+            Console.WriteLine("---------------------------------");
+
+            int count = 1;
+            foreach (var baseFile in baseDllFiles)
+            {
+                foreach (var targetFile in targetDllFiles)
+                {
+                    if (baseFile.Name == targetFile.Name)
+                    {
+                        try
+                        {
+                            var fileResult = CheckFile(baseFile.FullName, targetFile.FullName);
+                            Console.WriteLine($"{count++}. {baseFile.ToString()} : {((fileResult & ABITestResult.ACRRequired) == ABITestResult.ACRRequired ? "FAIL" : "PASS")}");
+                            directoryResult |= fileResult;
+                        }
+                        catch (Exception)
+                        {
+                            continue;
+                        }
+                    }
+                }
+            }
+            return directoryResult;
+        }
+
+        ABITestResult CheckChangedModule(FileInfo[] baseDllFiles, FileInfo[] targetDllFiles)
+        {
+            var result = ABITestResult.None;
+
+            var removedDllFiles = baseDllFiles.Where(b => !targetDllFiles.Any(t => t.Name == b.Name));
+            foreach (var file in removedDllFiles)
+            {
+                Console.WriteLine($"!! Missing DLL: {file.Name}");
+                result |= CheckFile(file.FullName);
+            }
+
+            var addedDllFiles = targetDllFiles.Where(t => !baseDllFiles.Any(b => b.Name == t.Name));
+            foreach (var file in addedDllFiles)
+            {
+                Console.WriteLine($"!! Added DLL : {file.Name}");
+                result |= CheckFile(file.FullName);
+            }
+
+            return result;
+        }
+
+        ABITestResult CheckFile(string file)
+        {
+            return CheckFile(file, string.Empty);
+        }
+
+        ABITestResult CheckFile(string baseFile, string targetFile)
+        {
+            ABITestResult result = ABITestResult.None;
+            IList<MemberInfo> changedAPIList;
+            try
+            {
+                if (targetFile == string.Empty)
+                {
+                    changedAPIList = _checker.CheckAssemblyFile(Assembly.LoadFrom(baseFile));
+                }
+                else
+                {
+                    changedAPIList = _checker.CheckAssemblyFile(Assembly.LoadFrom(baseFile), Assembly.LoadFile(targetFile));
+                }
+                var internalAPIList = new List<MemberInfo>();
+                for (int i = changedAPIList.Count - 1; i >= 0; i--)
```

Comment:
why not use 'foreach'?

Reply:
Removing the item of List inside the loop ruins the List.